contivctl/commands.go

```diff
@@ -0,0 +1,191 @@
+
+import (
+	"github.com/codegangsta/cli"
+)
+
+var tenantFlag = cli.StringFlag{
+	Name:  "tenant, t",
+	Value: "default",
+	Usage: "Name of the tenant to create the policy on.",
+}
+
+var Commands = []cli.Command{
+	{
+		Name:  "group",
+		Usage: "Endpoint Group manipulation tools",
+		Subcommands: []cli.Command{
+			{
+				Name:      "create",
+				Usage:     "Create an endpoint group",
+				ArgsUsage: "[group] [network]",
+				Flags: []cli.Flag{
+					tenantFlag,
+					cli.StringFlag{
+						Name:  "policy, p",
+						Usage: "Policy List (separated by commas)",
+					},
+				},
+				Action: CreateEndpointGroup,
+			},
+			{
+				Name:      "delete",
+				Usage:     "Delete an endpoint group",
+				ArgsUsage: "[group]",
+				Flags:     []cli.Flag{tenantFlag},
+				Action:    DeleteEndpointGroup,
+			},
+			{
+				Name:      "list",
+				Usage:     "List endpoint groups",
+				ArgsUsage: " ",
+				Flags: append([]cli.Flag{tenantFlag},
+					cli.BoolFlag{
+						Name:  "all, a",
+						Usage: "List endpoint groups for all tenants",
+					}),
+				Action: ListEndpointGroups,
+			},
+		},
+	},
+	{
+		Name:  "net",
+		Usage: "Network manipulation tools",
+		Subcommands: []cli.Command{
+			{
+				Name:      "list",
+				Usage:     "List networks",
+				ArgsUsage: " ",
+				Flags: append([]cli.Flag{tenantFlag}, cli.BoolFlag{
+					Name:  "all, a",
+					Usage: "Display all networks",
+				}),
+				Action: ListNetworks,
+			},
+			{
+				Name:      "delete",
+				Usage:     "Delete a network",
+				ArgsUsage: "[network]",
+				Flags:     []cli.Flag{tenantFlag},
+				Action:    DeleteNetwork,
+			},
+			{
+				Name:      "create",
+				Usage:     "Create a network",
+				ArgsUsage: "[network]",
+				Flags: []cli.Flag{
+					tenantFlag,
+					cli.BoolFlag{
+						Name:  "public, p",
+						Usage: "Indicate this network is public",
+					},
+					cli.StringFlag{
+						Name:  "encap, e",
+						Usage: "Packet tag",
+						Value: "vxlan",
+					},
+					cli.StringFlag{
+						Name:  "subnet, s",
+						Usage: "Subnet CIDR - REQUIRED",
+					},
+					cli.StringFlag{
+						Name:  "default-gw, g",
+						Usage: "Default Gateway - REQUIRED",
+					},
+				},
+				Action: CreateNetwork,
+			},
+		},
+	},
+	{
+		Name:  "rule",
+		Usage: "Rule manipulation tools",
+		Subcommands: []cli.Command{
+			{
+				Name:      "list",
+				Usage:     "List rules for a given tenant/policy",
+				ArgsUsage: "[policy]",
+				Flags:     []cli.Flag{tenantFlag},
+				Action:    ListRules,
+			},
+			{
+				Name:      "delete",
+				Usage:     "Delete an existing rule.",
+				ArgsUsage: "[policy] [rule id]",
+				Flags:     []cli.Flag{tenantFlag},
+				Action:    DeleteRule,
+			},
+			{
+				Name:      "add",
+				Usage:     "Add a new rule.",
+				ArgsUsage: "[policy] [rule id]",
+				Flags: []cli.Flag{
+					tenantFlag,
+					cli.IntFlag{
+						Name:  "priority, p",
+						Usage: "Priority Indicator",
+					},
+					cli.StringFlag{
+						Name:  "direction, d",
```

Comment:
Is there a way to specify the  values of direction 'in' and 'out' here? instead of doing inline validation i.e. enums in codegangsta/cli